calculators/referendum_expenditures_by_origin.rb

```diff
@@ -0,0 +1,101 @@
+class ReferendumExpendituresByOrigin
+  def initialize(candidates: [], ballot_measures: [], committees: [])
+    @ballot_measures = ballot_measures
+    @committees_by_filer_id =
+      committees.where('"Filer_ID" IS NOT NULL').index_by { |c| c.Filer_ID }
+  end
+
+  def fetch
+    totals = ActiveRecord::Base.connection.execute(<<-SQL)
+      SELECT "Measure_Number", "Sup_Opp_Cd", sum("Amount") AS total
+      FROM "efile_COAK_2016_E-Expenditure",
+        oakland_name_to_number
+      WHERE "Bal_Name" = "Measure_Name"
+      GROUP BY "Measure_Number", "Sup_Opp_Cd";
+    SQL
+    expenditures = ActiveRecord::Base.connection.execute(<<-SQL)
+      SELECT d."Measure_Number", d."Sup_Opp_Cd", A.locale, A.total
+      FROM
+        (SELECT distinct "Filer_ID", "Measure_Number", "Sup_Opp_Cd"
+          FROM "efile_COAK_2016_E-Expenditure",
+          oakland_name_to_number
+          WHERE "Bal_Name" IS NOT NULL
+          AND "Bal_Name" = "Measure_Name") d,
+        (SELECT "Filer_ID",
+        CASE
+          WHEN lower("Tran_City") = 'oakland' THEN 'Within Oakland'
+          WHEN upper("Tran_State") = 'CA' THEN 'Within California'
+          ELSE 'Out of State'
+        END as locale,
+        sum("Tran_Amt1") as total
+        FROM
+          (SELECT "Filer_ID", "Tran_City", "Tran_State", "Tran_Amt1"
+          FROM "efile_COAK_2016_A-Contributions"
+          UNION
+          SELECT CAST("Filer_ID" as VARCHAR(7)),
+          "Enty_City" as "Tran_City", "Enty_ST" as "Tran_State",
+          "Amount" as "Tran_Amt1"
+          FROM "efile_COAK_2016_497") x
+        WHERE "Filer_ID" IN (SELECT "Filer_ID"
+              FROM "efile_COAK_2016_E-Expenditure"
+              WHERE "Bal_Name" IS NOT NULL)
+        GROUP BY "Filer_ID", locale) A
+      WHERE d."Filer_ID" = A."Filer_ID"
+      ORDER BY d."Measure_Number", d."Sup_Opp_Cd";
```

Comment:
Now we're talkin'. I knew we'd get some > 20 line SQL queries in here before too long ;)